app/services/geographic_builder.rb

```diff
@@ -48,12 +48,13 @@ def build_place(node)
   def place_from_code(node)
     return [] unless node.code && node.source
 
-    code = node.code.sub(/[^\w-]/, '') # remove any punctuation (except dash).
+    code_with_dashes = node.code.gsub(/[^\w-]/, '') # remove any punctuation (except dash).
+    code_without_dashes = code_with_dashes.sub(/-+$/, '') # remove trailing dashes.
```

Comment:
Should this be in DSA? Shouldn't the cocina model only have valid codes?

Reply:
Depends on how you define "valid" I guess ...

I'm not opposed to a separate ticket in dor-services-app to consider.

Reply:
I would expect that valid means any marcgac codes to be on the list that LOC publishes.

Reply:
https://github.com/sul-dlss/dor-services-app/issues/2805